Macos build failure passing CharString to CFStringCreateWithCString

### Tested versions

Introduced in https://github.com/godotengine/godot/pull/101293

### System information

Godot v4.5.dev.mono (e278a6e8c) - macOS Sequoia (15.3.1) - Multi-window, 1 monitor - Metal (Forward+) - integrated Apple M2 (Apple8) - Apple M2 (8 threads)

### Issue description


Noticed local builds failing with `scons module_mono_enabled=yes dev_mode=yes dev_build=yes` (but not `scons dev_mode=yes dev_build=yes` for some reason) after rebasing earlier today and bisected to the above pr. I assume this instance was missed as it's behind `MACOS_ENABLED`. The build error is:
```
[ 56%] modules/mono/utils/macos_utils.cpp:41:26: error: no matching function for call to 'CFStringCreateWithCString'
   41 |         CFStringRef bundle_id = CFStringCreateWithCString(nullptr, p_bundle_id.utf8(), kCFStringEncodingUTF8);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFString.h:181:13: note: candidate function not viable: no known conversion from 'CharString' to 'const char *' for 2nd argument
  181 | CFStringRef CFStringCreateWithCString(CFAllocatorRef alloc, const char *cStr, CFStringEncoding encoding) CF_FORMAT_ARGUMENT(2);
      |             ^                                               ~~~~~~~~~~~~~~~~
```

### Steps to reproduce

Build on macos with c# support, see error from clang/scons

### Minimal reproduction project (MRP)

n/a

--- a/modules/mono/utils/macos_utils.cpp
+++ b/modules/mono/utils/macos_utils.cpp
@@ -38,7 +38,7 @@
 #import <CoreServices/CoreServices.h>
 
 bool macos_is_app_bundle_installed(const String &p_bundle_id) {
-	CFStringRef bundle_id = CFStringCreateWithCString(nullptr, p_bundle_id.utf8(), kCFStringEncodingUTF8);
+	CFStringRef bundle_id = CFStringCreateWithCString(nullptr, p_bundle_id.utf8().get_data(), kCFStringEncodingUTF8);
 	CFArrayRef result = LSCopyApplicationURLsForBundleIdentifier(bundle_id, nullptr);
 	CFRelease(bundle_id);
 

